metricflow/engine/models.py

```diff
@@ -107,6 +111,7 @@ def from_pydantic(
             is_partition=pydantic_dimension.is_partition,
             expr=pydantic_dimension.expr,
             label=pydantic_dimension.label,
+            _possible_custom_granularity_names=custom_granularity_names,
```

Comment:
I think using this param is kinda hacky and there should be a better option. My thought is that we should add a new attribute to the `Dimension` object to store the `entity_links`. Then we wouldn't need custom granularity names here at all. Instead the `granularity_free_qualified_name` would just use something like this logic:
`StructuredLinkableSpecName(element_name=self.name, entity_links=(entity_prefix,)).granularity_free_qualified_name`
And we could also add a property to resolve `Dimension.qualified_name` instead of having to pass that in.
Seems cleaner and it definitely seems useful to have the entity prefix in general!

Reply:
ooo, ok that makes sense!

Reply:
Sweet. Just a note here that we'll need to update MFS with this new property since I think we're using the `Dimension` object there!